### ISSUE TITLE
Segfault on "Refresh list"

### ISSUE THREAD

**Issue by progalski** [ShinjiruApp/Shinjiru#12]
Program starts up fine, requests access on AniList and says it's recieved the callback.
When trying to Refresh list from the AniList menu option though, the program has a segfault.
Compiled on Debian buster/sid using Qt 5.10.1 downloaded through the online installer.

Running the program through GDB gives this information, otherwise it just exits with the 'Segmentation fault' error.

Thread 1 "Shinjiru" received signal SIGSEGV, Segmentation fault.
QString::toLocal8Bit() const & (this=0x7fffc40042e8)
    at ../../Qt/5.10.1/gcc_64/include/QtCore/qstring.h:544
544	    { return toLocal8Bit_helper(isNull() ? nullptr : constData(), size()); }

**Comment by Kazakuri** [comment]
Does the stack trace go any deeper than `QString::toLocal8Bit()`? That function ends up getting called a ton as a result of the load process so it's not really telling of what's going wrong.

**Comment by progalski** [comment]
I'm not really a programmer, so I have no idea how to use gdb. I tried using valgrind instead to see if it gave more information. I attached the full logfile.
[shinjiru.log](https://github.com/ShinjiruApp/Shinjiru/files/2094460/shinjiru.log)

**Comment by Kazakuri** [comment]
It seems like it's failing to get the token completely, which is strange if it redirects you to the `Callback received. Feel free to close this page.` screen.

I pushed a commit that might help debug it in the log file.
Can you post the contents of the `~/.Shinjiru.log` file?

**Comment by progalski** [comment]
[.Shinjiru.log](https://github.com/ShinjiruApp/Shinjiru/files/2096003/default.Shinjiru.log)

**Comment by Kazakuri** [comment]
The log can't find a compatible version of OpenSSL, it seems like Qt 5.10 depends on at least OpenSSL 5.1.0. 

Try updating openssl via `sudo apt install openssl` since it [seems like](https://packages.debian.org/sid/openssl) sid distributes 1.1.0h.

**Comment by progalski** [comment]
openssl is already the newest version (1.1.0h-4).
Not really anything I can do about the openssl version it seems. I could check what libraries the files are linked to but I don't know which file links against openssl.

**Comment by Kazakuri** [comment]
Hm, if it's being linked at compile time then you'll probably need libssl up to date too. Maybe try that?

**Comment by progalski** [comment]
So, after googling a bit I found this: https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3001
Which then leads to this: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=797926
And the solution being: https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3001#issuecomment-309673453

TL;DR
Install ``libssl1.0-dev`` and recompile.
